libyul/YulStack.cpp

```diff
@@ -241,6 +245,33 @@ void YulStack::optimize(Object& _object, bool _isCreation)
 	);
 }
 
+void YulStack::reparse()
+{
+	yulAssert(m_parserResult);
+	yulAssert(m_charStream);
+
+	// NOTE: Without passing in _soliditySourceProvider, printed debug info will not include code
+	// snippets, but it does not matter - we'll still get the same AST after we parse it. Snippets
+	// are not stored in the AST and the other info that is (location, AST ID, etc) will still be present.
+	std::string source = print(nullptr /* _soliditySourceProvider */);
+
+	YulStack cleanStack(m_evmVersion, m_eofVersion, m_language, m_optimiserSettings, m_debugInfoSelection);
+	bool reanalysisSuccessful = cleanStack.parseAndAnalyze(m_charStream->name(), source);
```

Comment:
Using `YulStack` in compiler stack feels a bit like an overkill, but it seemed cleaner to me than refactoring the parsing methods to make them usable without overwriting what's in the current stack.